src/utils/safeHref.ts

```diff
@@ -0,0 +1,56 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+let useSafeHref = false
+
+export function setUseSafeHref(value: boolean) {
+    useSafeHref = value
+}
+
+export function getUseSafeHref(): boolean {
+    return useSafeHref
+}
+
+export function getURLInfo(url: string): URLInfo {
+    /**
+     * An array of the safely allowed url protocols
+     */
+    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:']
```

Comment:
```
The URL that the hyperlink points to. Links are not restricted to HTTP-based URLs — they can use any URL scheme supported by browsers:

Sections of a page with fragment URLs
- Pieces of media files with media fragments
- Telephone numbers with tel: URLs
- Email addresses with mailto: URLs
- While web browsers may not support other URL schemes, web sites can with registerProtocolHandler()
```

Is `data:` a safe protocol? We may rely on that for cases like `download`.

Reply:
Good question about data:

Reply:
Added per conversation in slack